plugin/trino-base-jdbc/src/main/java/io/trino/plugin/jdbc/BaseJdbcClient.java

```diff
@@ -555,6 +557,8 @@ protected JdbcOutputTableHandle createTable(ConnectorSession session, ConnectorT
             String remoteTargetTableName = identifierMapping.toRemoteTableName(identity, connection, remoteSchema, targetTableName);
             String catalog = connection.getCatalog();
 
+            verifyTableNameLength(connection.getMetaData(), remoteTargetTableName);
```

Comment:
We shouldn't need to do this here, so we shouldn't do this.

We generally trust that CREATE TABLE statement will do all the proper validation on the remote database's side.

If we cannot trust the database fully, why do we trust `databaseMetadata.getMaxTableNameLength`? It can be incorrect as well. 
(a driver has smaller user base than the database engine itself, and metadata methods may get even less scrutiny...)

Reply:
we need to do the check before sending the CREATE to PostgreSQL, otherwise we create a table with a wrong name.

for CREATE TABLE we can do the check in `io.trino.plugin.postgresql.PostgreSqlClient#createTable`.

is RENAME TO also affected?

Reply:
Yes, `RENAME TO` also truncates the name.

Reply:
that sounds like  we want to have a table validation hook in JdbcClient
and it would be called before CT and before Rename is issued to the remote system.

by default it would be empty, but for PostgreSQL we would add length checks.